flask-server/server.py

```diff
@@ -1,12 +1,120 @@
-from flask import Flask
+from flask import Flask, request, jsonify
+from flask_sqlalchemy import SQLAlchemy
+from flask_cors import CORS
+from datetime import datetime, timedelta
+import threading
 
 app = Flask(__name__)
+CORS(app) # Allows cross-origin requests from React frontend
+
+# Config SQLite database
+
+# Tells Flask-SQLAlchemy to use an SQLite database
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///products.db'
+# Set this to False to improve performance and avoid seeing unnecessary warnings.
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+# Set up the database
+db = SQLAlchemy(app)
+
+
+# Define Product model
+class Product(db.Model):
+  id = db.Column(db.Integer, primary_key=True)
+  name = db.Column(db.String(100), nullable=False)
+  expiration_date = db.Column(db.Date, nullable=False)
+  
+# Initialize the database
+with app.app_context():
+    db.create_all()
+
+# Make a route to add a product:
+@app.route('/add_product', methods=['POST'])
+def add_product():
+  data = request.get_json()
```

Comment:
We might need to add a validation layer when we add a new product so that we don't insert an invalid one into the database.

Reply:
I'm not too familiar with Flask though so there might not be an issue with this.